packages/zoltan2/src/algorithms/connectivity/Zoltan2_IceSheet.hpp

```diff
@@ -0,0 +1,306 @@
+//@HEADER
+#ifndef _ZOLTAN2_ICESHEET_HPP_
+#define _ZOLTAN2_ICESHEET_HPP_
+#include<unordered_set>
+#include<utility>
+#include<vector>
+#include<Zoltan2_GraphModel.hpp>
+#include<Zoltan2_Algorithm.hpp>
+#include<Zoltan2_PartitioningSolution.hpp>
+#include<Zoltan2_Util.hpp>
+#include<Zoltan2_TPLTraits.hpp>
+#include<Zoltan2_IceProp.hpp>
+#include<Zoltan2_AlltoAll.hpp>
+
+/*! \file Zoltan2_IceSheet.hpp
+ * \brief The user interface for ice-sheet connectivity functionality.
+ *
+ * This file contains the entry point for the ice sheet connectivity
+ * label propagation algorithm. 
+*/
+
+namespace Zoltan2{
+  /*! Utility function for detecting degenerate features of ice sheets
+   *  
+   *  \param (input) problemComm A Teuchos::RCP<const Teuchos::Comm<int>> representing the communicator for the problem.
+   *  \param (input) adapter A Teuchos::RCP<const GraphAdapter<>> representing the structure of the ice sheet mesh's bottom layer.
+   *  \param (input) basalFriction An ArrayView of boolean flags for each local vertex with true representing a grounded vertex and false representing a floating vertex.
+   *                        These indicies must be consistent with the local vertex identifiers for the input graph.
+   *  \param (input) boundary_edges An ArrayView of global vertex identifiers representing all of the edges that the current process knows about which are on the boundary between the ice and the water. 
+   *                        The IDs must be present in the overlap map of the current process if the ID is not owned by this process. Edges should not be specified more than once.
```

Comment:
Does "IDs must be present in the overlap map of the current process" mean that a process can handle a boundary edge between a ghost vertex and a vertex on another process?  Or must at least one vertex of each edge be owned by the process?

Regarding "Edges should not be specified more than once":  Say proc 0 owns ID A, and proc 1 owns ID B.  Does proc 0 give edge A->B and proc 1 give edge B->A?  Or is that an error?

Reply:
"IDs must be present in the overlap map of the current process" means that if there is an ID in the boundary edges, the current process must know which process owns that ID. This constraint was made from my understanding of Mauro's explanation of which boundary edges are easy to provide, so this should be what Mauro is expecting. If there is an edge from a ghost vertex to an unknown vertex (a vertex whose owner is unknown), the current process may not be able to easily send the edge to both of the processes that need it.

"Edges should not be specified more than once" is purely a local constraint. Globally, edges may be specified more than once, but I use a set to ensure that remote boundary edges are not counted more than once, as they could exist on more than one process. I can add this to the comment to make it more clear.

Reply:
So to answer your second question, A->B on proc 0 and B->A on proc 1 is fine.